common/src/main/java/org/wso2/testgrid/common/config/InfrastructureConfig.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.wso2.testgrid.common.config;
+
+import org.apache.commons.collections4.ListUtils;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Defines a model object for a InfrastructureConfig.
+ *
+ * @since 1.0.0
+ */
+public class InfrastructureConfig implements Serializable {
+
+    private static final long serialVersionUID = -1660815137752094462L;
+
+    private IACProvider iacProvider;
+    private InfrastructureProvider infrastructureProvider;
+    private ContainerOrchestrationEngine containerOrchestrationEngine;
+    private Properties parameters = new Properties();
+    private List<Provisioner> provisioners;
+
+    /**
+     * Defines the infrastructure provider types.
+     *
+     * @since 1.0.0
+     */
+    public enum IACProvider {
+        CLOUDFORMATION("CloudFormation"),
+        TERRAFORM("Terraform"),
+        None("None");
+        private final String iacProvider;
+
+        IACProvider(String iacProvider) {
+            this.iacProvider = iacProvider;
+        }
+
+        @Override
+        public String toString() {
+            return this.iacProvider;
+        }
+    }
+
+    /**
+     * Defines the infrastructure provider types.
+     *
+     * @since 1.0.0
+     */
+    public enum InfrastructureProvider {
+        AWS("AWS"),
+        OPENSTACK("OpenStack"),
+        GCP("GCP"),
+        SHELL("SHELL"),
```

Comment:
Is SHELL appropriate here among other cloud providers?

Reply:
No, we need to improve this logic. Created https://github.com/wso2/testgrid/issues/500 to track.

This was [existing logic](https://github.com/wso2/testgrid/pull/495/files/9682d888d42bf4ef2b50034c0a3edc9dd8f883cf#diff-83442d29c87f411acc105a9759934c6fL215) that was brought into a new class.